docs/devel/node-performance-testing.md

```diff
@@ -0,0 +1,146 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.1/docs/devel/node-performance-testing.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Measuring Node Performance
+
+This document outlines the issues and pitfalls of measuring Node performance, as well as the tools
+available.
+
+## Cluster Set-up
+
+There are lots of factors which can affect node performance numbers, so care must be taken in
+setting up the cluster to make the intended measurements. In addition to taking the following steps
+into consideration, it is important to document precisely which setup was used. For example,
+performance can vary wildly from commit-to-commit, so it is very important to document which commit
+or version of Kubernetes was used, which Docker version was used, etc.
+
+### Addon pods
+
+Be aware of which addon pods are running on which nodes. By default Kubernetes runs 8 addon pods,
+plus another 2 per node (`fluentd-elasticsearch` and `kube-proxy`) in the `kube-system`
+namespace. The addon pods can be disabled for more consistent results, but doing so can also have
+performance implications.
+
+For example, Heapster polls each node regularly to collect stats data. Disabling Heapster will hide
+the performance cost of serving those stats in the Kubelet.
+
+#### Disabling Add-ons
+
+Disabling addons is simple. Just ssh into the Kubernetes master and move or delete the addon from
+`/etc/kubernetes/addons/`.
```

Comment:
A link to the add-ons may be useful in case this document becomes outdated: https://github.com/kubernetes/kubernetes/tree/master/cluster/addons